core/src/main/java/org/mule/runtime/core/internal/processor/strategy/ComponentProcessor.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.core.internal.processor.strategy;
+
+import org.mule.runtime.core.api.processor.ReactiveProcessor;
+
+/**
+ * Specialization of {@link ReactiveProcessor} that allows to perform certain optimizations when the processor is not blocking, as
```

Comment:
I think this contract needs to be better defined. If the purpose is to allow for optimizations, then the interface name should reflect that an execution type is what's being modeled. If what this is modeling is the execution of a mule component, then the jdoc should reflect that and execution optimizations is only ONE aspect of that

Reply:
Second option:
Specialization of {@link ReactiveProcessor} that allows execution of a mule component taking into account its processing strategy when it has to be decoupled from a policy (see MULE-17939). This allows performing certain optimizations when the processor is not blocking, as indicated by its {@link #isBlocking()} method.